diff -Naur fence-agents-3.1.5.orig/fence/agents/lib/fencing.py.py fence-agents-3.1.5/fence/agents/lib/fencing.py.py
--- fence-agents-3.1.5.orig/fence/agents/lib/fencing.py.py	2015-02-10 16:30:33.455431070 -0800
+++ fence-agents-3.1.5/fence/agents/lib/fencing.py.py	2015-02-10 16:31:34.447202203 -0800
@@ -808,7 +808,7 @@
 		for o in outlets.keys():
 			(alias, status) = outlets[o]
 			if options["-o"] != "monitor":
-				print o + options["-C"] + alias	
+				print o + options["-C"] + alias	+ options["-C"] + status
 		return
 
 	status = get_power_fn(tn, options)
