diff -ru fence-agents-3.1.5/fence/agents/apc_snmp/fence_apc_snmp.py fence-agents-3.1.5-patch0002/fence/agents/apc_snmp/fence_apc_snmp.py
--- fence-agents-3.1.5/fence/agents/apc_snmp/fence_apc_snmp.py	2015-02-10 11:46:21.001593054 -0800
+++ fence-agents-3.1.5-patch0002/fence/agents/apc_snmp/fence_apc_snmp.py	2015-02-10 11:39:56.363920890 -0800
@@ -69,12 +69,25 @@
 	turn_off=2
 	has_switches=False
 
+class ApcMS6:
+	# Master Switch with 6.x firmware
+	status_oid = '.1.3.6.1.4.1.318.1.1.4.4.2.1.3.%d'
+	control_oid = '.1.3.6.1.4.1.318.1.1.12.3.3.1.1.4.%d'
+	outlet_table_oid = '1.3.6.1.4.1.318.1.1.4.4.2.1.4'
+	ident_str = "APC Master Switch with firmware v6.x"
+	state_on = 1
+	state_off = 2
+	turn_on = 1
+	turn_off = 2
+	has_switches = False
+
 ### FUNCTIONS ###
 def apc_set_device(conn,options):
 	global device
 
 	agents_dir={'.1.3.6.1.4.1.318.1.3.4.5':ApcRPDU,
 		    '.1.3.6.1.4.1.318.1.3.4.4':ApcMSP,
+		    '.1.3.6.1.4.1.318.1.3.4.6':ApcMS6,
 		    None:ApcMS}
 
 	# First resolve type of APC
