--- tests/regressiontests/servers/tests.py.dist	2013-04-01 10:46:10.000000000 -0700
+++ tests/regressiontests/servers/tests.py	2013-04-01 10:45:28.000000000 -0700
@@ -150,6 +150,8 @@
             raise Exception("The line above should have raised an exception")
         except exception:
             pass
+        finally:
+            super(LiveServerAddress, cls).tearDownClass()
 
     def test_test_test(self):
         # Intentionally empty method so that the test is picked up by the
