diff --git a/mod_proxy_wstunnel.c.warnings b/mod_proxy_wstunnel.c
index cfcaf57..b2170bc 100644
--- a/mod_proxy_wstunnel.c.warnings
+++ b/mod_proxy_wstunnel.c
@@ -74,7 +74,6 @@ PROXY_DECLARE(int) ap_proxy_create_hdrbrgd(apr_pool_t *p,
     apr_bucket *e;
     int do_100_continue;
     conn_rec *origin = p_conn->connection;
-    proxy_dir_conf *dconf = ap_get_module_config(r->per_dir_config, &proxy_module);
 
     /*
      * To be compliant, we only use 100-Continue for requests with bodies.
@@ -716,7 +715,9 @@ static void ap_proxy_http_register_hook(apr_pool_t *p)
     proxy_hook_canon_handler(proxy_wstunnel_canon, NULL, NULL, APR_HOOK_FIRST);
 }
 
+#ifdef APLOG_USE_MODULE
 APLOG_USE_MODULE(proxy_wstunnel);
+#endif
 module AP_MODULE_DECLARE_DATA proxy_wstunnel_module = {
     STANDARD20_MODULE_STUFF,
     NULL,                       /* create per-directory config structure */
