--- tools/gyp/pylib/gyp/generator/make.py.orig	2017-01-05 16:30:47.000000000 -0500
+++ tools/gyp/pylib/gyp/generator/make.py	2017-03-25 12:41:51.197566549 -0400
@@ -135,13 +135,26 @@
 # This is the replacement character.
 SPACE_REPLACEMENT = '?'
 
-
 LINK_COMMANDS_LINUX = """\
+# Macro for invoking a command on groups of 1000 words at a time
+# (analogous to xargs(1)).  The macro invokes itself recursively
+# until the list of words is depleted.
+#
+# Usage: $(call xargs,COMMAND,LIST)
+#
+# COMMAND should be a shell command to which the words will be
+# appended as arguments in groups of 100.
+define xargs
+$(1) $(wordlist 1,100,$(2))
+$(if $(word 101,$(2)),$(call xargs,$(1),$(wordlist 101,$(words $(2)),$(2))))
+endef
+
 quiet_cmd_alink = AR($(TOOLSET)) $@
 cmd_alink = rm -f $@ && $(AR.$(TOOLSET)) crs $@ $(filter %.o,$^)
 
 quiet_cmd_alink_thin = AR($(TOOLSET)) $@
-cmd_alink_thin = rm -f $@ && $(AR.$(TOOLSET)) crsT $@ $(filter %.o,$^)
+#cmd_alink_thin = rm -f $@ && $(call xargs,$(AR.$(TOOLSET)) crsT $@ $(1),$(filter %.o,$^)) 
+cmd_alink_thin = rm -f $@ && $(AR.$(TOOLSET)) crsT $@ $(filter %.o,$^)
 
 # Due to circular dependencies between libraries :(, we wrap the
 # special "figure out circular dependencies" flags around the entire
@@ -260,7 +260,8 @@
 abs_srcdir := $(abspath $(srcdir))
 
 # The name of the builddir.
-builddir_name ?= %(builddir)s
+#builddir_name ?= %(builddir)s
+builddir_name ?= .
 
 # The V=1 flag on command line makes us verbosely print command lines.
 ifdef V
